### DISCUSSION TITLE
Funding fees on perpetual contract

### DISCUSSION THREAD

**Discussion by jiro1072** [kernc/backtesting.py#868] [Q&A]
Hello,

When we [run a backtest](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Backtest.run) (with ccxt binance as a client for example) on a perpetual contract ( like BTCPERP), Does the result take into account the funding rate?
If no, is there a possibility to add it into the fees?

Thanks !

**Comment by kernc** [comment]
> Does the result take into account the funding rate?

If you mean financing for the implied margin trading, it does not.

You can apply additional financing costs post-backtest based on `stats['_trades']`, duration of trades, your trading leverage and interest rate. Then you can pass modified `stats['trades']` df into [`backtesting.lib.compute_stats()`](https://kernc.github.io/backtesting.py/doc/backtesting/lib.html#backtesting.lib.compute_stats).
